nyaa/search.py

```diff
@@ -3,15 +3,99 @@
 import math
 import json
 import shlex
+from datetime import datetime, timedelta
 
 from nyaa import app, db
-from nyaa import models
+from nyaa import models, torrents
 
 import sqlalchemy_fulltext.modes as FullTextMode
 from sqlalchemy_fulltext import FullTextSearch
 from elasticsearch import Elasticsearch
 from elasticsearch_dsl import Search, Q
 
+class ElasticTorrentStatsAccessor:    
+    def __init__(self, elasticResult):
+        self.seed_count = elasticResult.seed_count
+        self.leech_count = elasticResult.leech_count
+        self.download_count = elasticResult.download_count
+
+class ElasticTorrentAccessor:
+    knownAttributes = ['id', 'has_torrent', 'display_name', 'filesize', 'anonymous', 'trusted', 'remake', 'complete', 'hidden', 'deleted']
+
+    def __init__(self, elasticResult):
+        self.elastic = elasticResult
+
+    def __getattr__(self, name):
+        # Passthru for attributes that are the same
+        if name in self.knownAttributes:
+            return self.elastic[name]
+
+        # Safetynet: throw if an attribute is accessed which we don't implement    
+        raise Exception("Unknown attribute {}".format(name))
+
+    @property
+    def created_time(self):
+        return datetime.strptime(self.elastic.created_time, '%Y-%m-%dT%H:%M:%S')
```

Comment:
What's the purpose of all these `@property` attributes instead of just assigning them directly in the `__init__`?
```python
self.create_time = datetime.strptime(self.elastic.created_time, '%Y-%m-%dT%H:%M:%S')
```
They require evaluation on every call to the attribute, which only makes sense if `self.elastic` is expected to change between every call.  

If they are intended to be lazily evaluated, then the calculated value should be cached in a protected attribute.  An example for `created_time` follows:
```python
    @property
    def created_time(self):
        attribute = '_created_time'
        result = None
        try:
            result = getattr(self, attribute)
        except AttributeError:
            result = datetime.strptime(self.elastic.created_time, '%Y-%m-%dT%H:%M:%S')
            setattr(self, attribute, result)
        finally:
            return result
```